documentation/swagger-checklist.md

```diff
@@ -43,82 +45,236 @@ This document defines a Swagger checklist for Microsoft Azure, along with clarif
 
-&#x25a2; *M2009*: A collection model in Swagger MUST support adding more than one element to the collection.
+- [ ] **M2008**: Error definitions *MUST* be abstracted, modeled and reused.
 
-&#x25a2; *M2010*: Operations that are exposed through private previews MUST not be described in Swaggers in the [public GitHub repository](https://github.com/Azure/azure-rest-api-specs).
+- [ ] **M2009**: A Swagger spec *MUST NOT* define operations or properties or parameters for functionalities that are not currently supported by the service in the given api-version. The primary goal of the spec is that it *MUST* correctly and completely represent the underlying REST API.
 
+- [ ] **M2010**: A collection model in Swagger *MUST* support adding more than one element to the collection. TODO: Ask for clarity.
 
-## Operational - Swagger Checklist ##
+- [ ] **M2011**: Operations that are exposed through private previews *MUST NOT* be described in the [public GitHub repository](https://github.com/Azure/azure-rest-api-specs). They *MUST* be described in the [private GitHub repository](https://github.com/Azure/azure-rest-api-specs-pr) instead.
+
+- [ ] **M2012**: The value of the ["x-ms-client-name"](https://github.com/Azure/autorest/tree/master/docs/extensions#x-ms-client-name) and ["x-ms-discriminator-value"](https://github.com/Azure/autorest/tree/master/docs/extensions#x-ms-discriminator-value) *MUST NOT* be the same as the model property or the model name, otherwise the purpose is defeated.
+
+- [ ] **M2013**: For specifications describing the Management Plane of ARM, "subscriptionId" & "api-version" *MUST* always be defined in the global parameters section.
+
+- [ ] **M2014**: For a Given Resource, GET/PUT/PATCH *MUST* return the same "Resource" Model.
+
+- [ ] **M2015**: For PATCH operation, every property in the model for the request body *MUST* be optional and the property *MUST NOT* provide any default value. The absence of defualt values applies recursively to complex properties in the model definition.
+
+- [ ] **M2016**: The model definition for the body parameter and the response must be the same for a PUT operation. This ensures that the same entity will be reusable between GET and PUT.
+
+- [ ] **M2017**: A parameter or a model property having an `"enum": []` constraint, *MUST* have an  ["x-ms-enum"](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-extensions.md#x-ms-enum) extension and the extension's "modelAsString" property *MUST* be set to `false` where applicable.
+
+- [ ] **M2018**: Every "Resource" Model *MUST* be tagged with ["x-ms-azure-resource"](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-extensions.md#x-ms-azure-resource). This will indicate Autorest to make the Resource model inherit from the Resource definition in the client runtime.
```

Comment:
Whenever we are referring to ARM entities, we should adhere to the ARM terminology (and ideally refer to the correct specifications/recommendations there). I believe that what we mean by "Resource" Model is a tracked resource, correct? If so, there are already requirements w. regards to which properties are required to be required vs. optional.